metadata-ingestion/src/datahub/ingestion/source/usage/snowflake_usage.py

```diff
@@ -393,10 +393,10 @@ def is_object_valid(obj: Dict[str, Any]) -> bool:
                 event_dict["query_start_time"]
             ).astimezone(tz=timezone.utc)
 
-            if not event_dict["email"] and self.config.email_domain:
-                if not event_dict["user_name"]:
+            if not event_dict["email"]:
+                if not event_dict["user_name"] or not self.config.email_domain:
                     logging.warning(
-                        f"The user_name is missing from {event_dict}. Skipping ...."
+                        f"A user cannot be generated from {event_dict}. Either the user name or email domain is missing. Skipping ...."
```

Comment:
Since the email domain is not supposed to be part of the event_dict, this message is misleading. 
Also suggest different ways to compute the values, so that the message is more specific -> is the user-name missing or is the email_domain not configured? Is there a way to do this without driving up complexity of the code?